Ensure tests clean their tempfiles

grahamwhaley opened the pull request:
A number of tests were not cleaning up after themselves, which left
a bunch of files in the TMPDIR.
Ensure we remove all the tmpfiles and dirs we create.

sboeuf commented:
@grahamwhaley CI error:
```
ERROR: No "Fixes" found
```

grahamwhaley commented:
Hmm, but it is in the second commit.... let me check....

chavafg commented:
I don't think the `Fixes: ` is needed for every commit, but I see something weird here: the checkcommits tool says that it found 3 commits, while the PR contains 4.

 ```
Running checkcommits version 0.0.1 (commit 805092aeeeb8897b355bbcb2b174ff3d5dda59ae)
Defaulting commit to HEAD
Defaulting branch to master
Found 3 commits between commit HEAD and branch master
ERROR: No "Fixes" found
ERROR: checkcommits failed. See the document below for help on formatting
commits for the project.
```

bug in the `checkcommits` tool?

grahamwhaley commented:
Err.... **no you do not**.
From: https://github.com/kata-containers/tests/tree/master/cmd/checkcommits#overview
`checking for at least one "Fixes #XXX" entry in the commits`
and from https://github.com/kata-containers/community/blob/master/CONTRIBUTING.md#general-format:
```
A "Fixes #XXX" comment listing the GitHub issue this change resolves.
This comment is required for the main patch in a sequence. See the following examples.
```

I've never 'Fixes' every commit - just normally the last one in the sequence :-)
Maybe that is the thing here - it is not the first or last commit - but, I've never seen that fail before.
I'll have a look

grahamwhaley commented:
@chavafg - yeah, hmm, that could be a bug in the tool or CI - there are indeed 4 patches. Heh, looks like we are going to have some checkcommits fun with no @jodh-intel :-)

grahamwhaley commented:
OK, I think I know what happened!
Over on #475 I also fixed one of the issues in one of the commits here - so during the rebase/merge on the CI I think that commit gets dropped - hence the CI only sees 3 commits (the same happened to me here when I rebased to master - took a little bit of figuring out why I only then had 3 commits!).
It also happened to have been the commit with the `Fixes` in it :-) Lesson learnt - something to look out for in the future...
Rebased, reworeded, re-pushed...

grahamwhaley commented:
Looks like static checks are fine, but the CIs suffered from the (unrelated I believe) https://github.com/kata-containers/tests/issues/513
Rather than re-kick them all, will wait for any input from @chavafg on if that is a transient, persistent or sporadic failure mode.

egernst commented:
@grahamwhaley @chavafg  - we still waiting for a new rekick?

grahamwhaley commented:
Yep, let me do that (probably with a rebase) - let's see how the CI is feeling..

bergwolf commented:
LGTM!

Looks like travis-ci is having issues with linux-ppc64le platform. Let's merge ignoring it since all the other CIs have passed.

sboeuf commented:
@grahamwhaley please backport this PR to stable branches if not present.

sboeuf commented:
Nevermind, it's already part of the stable branches.
